src/pages/SearchPage.js

```diff
@@ -32,82 +31,73 @@ const propTypes = {
 
     /** All reports shared with the user */
     reports: PropTypes.objectOf(reportPropTypes),
-
-    /** Window Dimensions Props */
-    ...windowDimensionsPropTypes,
-
-    ...withLocalizePropTypes,
 };
 
-const defaultProps = {
-    betas: [],
-    personalDetails: {},
-    reports: {},
-};
+function SearchPage({betas = [], personalDetails = {}, reports = {}}) {
+    // Data for initialization (runs only on the first render)
+    const {recentReports: initialRecentReports, personalDetails: initialPersonalDetails, userToInvite: initialUserToInvite} = useMemo(() => OptionsListUtils.getSearchOptions(reports, personalDetails, '', betas), []);
 
-class SearchPage extends Component {
-    constructor(props) {
-        super(props);
+    const [searchValue, setSearchValue] = useState('')
+    const [searchOptions, setSearchOptions] = useState({
+        recentReports: initialRecentReports,
+        personalDetails: initialPersonalDetails,
+        userToInvite: initialUserToInvite
+    });
 
-        Timing.start(CONST.TIMING.SEARCH_RENDER);
-        Performance.markStart(CONST.TIMING.SEARCH_RENDER);
+    const {translate} = useLocalize();
 
-        this.searchRendered = this.searchRendered.bind(this);
-        this.selectReport = this.selectReport.bind(this);
-        this.onChangeText = this.onChangeText.bind(this);
-        this.debouncedUpdateOptions = _.debounce(this.updateOptions.bind(this), 75);
+    const updateOptions = useCallback(() => {
+        const {recentReports: localRecentReports, personalDetails: localPersonalDetails, userToInvite: localUserToInvite} = OptionsListUtils.getSearchOptions(
+            reports,
+            personalDetails,
+            searchValue.trim(),
+            betas,
+        );
 
-        const {recentReports, personalDetails, userToInvite} = OptionsListUtils.getSearchOptions(props.reports, props.personalDetails, '', props.betas);
+        setSearchOptions({recentReports: localRecentReports, personalDetails: localPersonalDetails, userToInvite: localUserToInvite})
+    }, [reports, personalDetails, searchValue, betas])
 
-        this.state = {
-            searchValue: '',
-            recentReports,
-            personalDetails,
-            userToInvite,
-        };
-    }
+    const debouncedUpdateOptions = useCallback(_.debounce(updateOptions, 75), [updateOptions]);
 
-    componentDidUpdate(prevProps) {
-        if (_.isEqual(prevProps.reports, this.props.reports) && _.isEqual(prevProps.personalDetails, this.props.personalDetails)) {
-            return;
-        }
-        this.updateOptions();
-    }
+    useEffect(() => {
+        Timing.start(CONST.TIMING.SEARCH_RENDER);
+        Performance.markStart(CONST.TIMING.SEARCH_RENDER);
+    },[])
 
-    onChangeText(searchValue = '') {
-        this.setState({searchValue}, this.debouncedUpdateOptions);
-    }
+    useEffect(() => {
+        debouncedUpdateOptions()
+    }, [searchValue, debouncedUpdateOptions])
```

Comment:
Actually, `searchValue` not necessary here. 
`useEffect` has `debouncedUpdateOptions` has `updateOptions` has `searchValue`. BTW, if you want to simplify this chain, you can pass `searchValue` as an argument to `updateOptions`.
```suggestion
    }, [debouncedUpdateOptions])
```